tool/tsh/common/access_request_test.go

```diff
@@ -202,23 +250,31 @@ func TestAccessRequestSearch(t *testing.T) {
 			t.Parallel()
 			homePath, _ := mustLoginLegacy(t, s, tc.args.teleportCluster)
 			captureStdout := new(bytes.Buffer)
+			args := append([]string{
+				"--insecure",
+				"request",
+				"search",
+				fmt.Sprintf("--kind=%s", tc.args.kind),
+			},
+				tc.args.extraArgs...,
+			)
+			if tc.args.format != "" {
+				args = append(args, fmt.Sprintf("--format=%s", tc.args.format))
+			}
 			err := Run(
 				context.Background(),
-				append([]string{
-					"--insecure",
-					"request",
-					"search",
-					fmt.Sprintf("--kind=%s", tc.args.kind),
-				},
-					tc.args.extraArgs...,
-				),
+				args,
 				setCopyStdout(captureStdout),
 				setHomePath(homePath),
 			)
 			require.NoError(t, err)
-			// We append a newline to the expected output to esnure that the table
-			// does not contain any more rows than expected.
-			require.Contains(t, captureStdout.String(), tc.wantTable()+"\n")
+			if tc.args.format == "json" || tc.args.format == "yaml" {
```

Comment:
Same thing here. Instead of asserting on the exact output, use `require.JSONEq` or `require.YAMLEq` to assert that the result is equivalent to the expected value.